Tone Mode fixes - better support for TSQL and Cross modes

michael-muir opened the pull request:
Fixed rx and tx CTCSS offsets for TSQL tone modes.   Added partial support for Cross tone modes including Tone->Tone, Tone->DTCS and DTCS->Tone.

michael-muir committed:
Tone Mode fixes - better support for TSQL and Cross modes

Fixed rx and tx CTCSS offsets for TSQL tone modes.   Added partial support for Cross tone modes including Tone->Tone, Tone->DTCS and DTCS->Tone.

michael-muir commented:
Fixed rx and tx CTCSS offsets for TSQL tone modes. Added partial support for Cross tone modes including Tone->Tone, Tone->DTCS and DTCS->Tone.    To get full Cross mode support for the less frequent cross modes may require reworking the encoding and decoding.

Imran-imtiaz48 reviewed (commented):
Fixed CTCSS RX and TX tone offsets for TSQL modes. Added partial support for Cross Tone configurations including Tone to Tone, Tone to DTCS, and DTCS to Tone. Full support for the remaining Cross Tone variants may require a deeper update to the current encoding and decoding logic.

michael-muir committed:
More tmode fixes

Still buggy, there are different behaviors between reading the radio, data file or pasting entries that complicates the implementation.
